### ISSUE TITLE
args: count value doesn't affect most_confident proc block output

### ISSUE THREAD

**Issue by Mohit0928** [hotg-ai/rune#331]
```
most_confident:
    proc-block: "hotg-ai/rune#proc_blocks/most_confident_indices"
    inputs:
      - efficientdet.1
    outputs:
      - type: U32
        dimensions: [25]
    args:
      count: 0
```
In the above snippet of Runefile.yml, I have set `count` to 0, and it will still give the output. I can change it to anything, and it will create no effect on the output. The output is decided by  `outputs: dimensions`. In this case, it will give a [1,25] dimension output. I think we can remove the `count` from the `most_confident proc-block`

**Comment by Michael-F-Bryan** [comment]
Can you paste the contents of the generated `lib.rs`? 

The `most-confident-indices` proc block [uses the `count` parameter](https://github.com/hotg-ai/rune/blob/174ffae7e18a9942dde41ccce46596af4412f83e/proc-blocks/most_confident_indices/src/lib.rs#L49), so I'm guessing #300 might have introduced a bug which doesn't set the parameter in the first place.

**Comment by Mohit0928** [comment]
I took our commonly used [food](https://github.com/hotg-ai/test-runes/blob/master/image/food/Runefile.yml) example. I changed the `most_confident : args: count: 3` to `most_confident : args: count: 0` keeping `most_confident: outputs: dimensions: [3]` same in both cases and then build the Rune. I ran the inference on the [ramen.jpg](https://github.com/hotg-ai/test-runes/blob/master/image/food/assets/ramen.jpg). In both cases, I got the below output:
```
[{"type_name":"&str","channel":2,"elements":["Ramen","Udon","Pork_Bones"],"dimensions":[3]},{"type_name":"u32","channel":2,"elements":[1445,234,85],"dimensions":[3]}]
```

**Comment by Michael-F-Bryan** [comment]
Can you paste the actual Runefile? Writing everything inline like that is hard to understand and misses a bunch of details.

**Comment by Mohit0928** [comment]
@Michael-F-Bryan, here is the `Runefile.yml`.
```
version: "1"
image: runicos/base
pipeline:
  image:
    capability: IMAGE
    outputs:
      - type: U8
        dimensions: [1,192,192,3]
    args:
      width: 192
      height: 192
      pixel_format: "@PixelFormat::RGB"
  food:
    model: "./food.tflite"
    inputs:
      - image
    outputs:
      - type: U8
        dimensions: [1,2024]
  most_confident:
    proc-block: "hotg-ai/rune#proc_blocks/most_confident_indices"
    inputs:
      - food
    outputs:
      - type: U32
        dimensions: [3]
    args:
      count: 0
  label:
    proc-block: "hotg-ai/rune#proc_blocks/label"
    inputs:
      - most_confident
    outputs:
      - type: UTF8
        dimensions:
          - 3
    args:
      labels:
        - __background__
    ..................................
            - Kondowole
  serial:
    out: SERIAL
    inputs:
      - label
   ```
  I'm not using any custom proc-block in the above `Runefile.yml`.  As you can see in the Runefile.yml, I have set the `most_confident` args -> count to 0 instead of 3. It will not create any impact on the output generated by running `rune run model.rune .....`
 
 I take a look at the `most_confident_indices` proc-block. This part of the [code](https://github.com/hotg-ai/rune/blob/master/proc-blocks/most_confident_indices/src/lib.rs#:~:text=%7D-,impl%20HasOutputs%20for%20MostConfidentIndices%20%7B,%7D,-%23%5Bcfg(test)%5D) is using output dimension to decide the output dimension and making it independent of `count`. If I comment on this part then the output dimension will be equal to `count`

**Comment by Mohit0928** [comment]
@Michael-F-Bryan, I have fixed this issue. Closing it now.

**Comment by Michael-F-Bryan** [comment]
@Mohit0928, let's reopen this ticket and let the PR that resolves it automatically close the ticket when it gets merged.

**Comment by Michael-F-Bryan** [comment]
I believe this was actually resolved as part of #337, but we thought it was still broken because we were using an old version of `rune` for testing. 

@Mohit0928, feel free to re-open this ticket if it is still broken.
